pandas/plotting/_core.py

```diff
@@ -308,10 +308,21 @@ def _setup_subplots(self):
 
         axes = _flatten(axes)
 
-        if self.logx or self.loglog:
+        valid_log = (False, True, 'sym', None)
+
+        for i in (self.logx, self.logy, self.loglog):
+            if i not in valid_log:
+                raise ValueError("Valid inputs are boolean, None and 'sym'.")
```

Comment:
expanding on @jreback's comment to use sets, you could just check the emptiness of a set difference:
```python
valid_log = {False, True, 'sym', None}
input_log = {self.logx, self.logy, self.loglog}
if input_log - valid_log:
    raise ValueError("Valid inputs are boolean, None and 'sym'.")